modules/dxp/apps/segments/segments-experiment-test/src/testIntegration/java/com/liferay/segments/experiment/web/internal/portlet/action/test/GetDataMVCResourceCommandTest.java

```diff
@@ -0,0 +1,197 @@
+	}
+
+	@Test
+	public void testGetProps() throws Exception {
+		Layout layout = LayoutTestUtil.addTypeContentLayout(_group);
+
+		MockHttpServletRequest mockHttpServletRequest =
+			new MockHttpServletRequest();
+
+		MockLiferayResourceRequest mockLiferayResourceRequest =
+			new MockLiferayResourceRequest();
+
+		MockLiferayResourceResponse mockLiferayResourceResponse =
+			new MockLiferayResourceResponse();
+
+		ThemeDisplay themeDisplay = new ThemeDisplay();
+
+		themeDisplay.setCompany(_company);
+
+		themeDisplay.setLanguageId(_group.getDefaultLanguageId());
+		themeDisplay.setLayout(layout);
+		themeDisplay.setLocale(
+			LocaleUtil.fromLanguageId(_group.getDefaultLanguageId()));
+		themeDisplay.setLocale(_locale);
+		themeDisplay.setPermissionChecker(
+			PermissionThreadLocal.getPermissionChecker());
+		themeDisplay.setPortalDomain("localhost");
+		themeDisplay.setPortalURL(_company.getPortalURL(_group.getGroupId()));
+		themeDisplay.setSecure(true);
+		themeDisplay.setServerName("localhost");
+		themeDisplay.setServerPort(8080);
+		themeDisplay.setSiteGroupId(_group.getGroupId());
+
+		mockHttpServletRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, themeDisplay);
+
+		mockLiferayResourceRequest.setAttribute(
+			PortletServlet.PORTLET_SERVLET_REQUEST, mockHttpServletRequest);
+
+		String url = "http://localhost:8080";
+
+		mockLiferayResourceRequest.setParameter("backURL", url);
+		mockLiferayResourceRequest.setParameter("redirect", url);
+
+		mockLiferayResourceRequest.setParameter(
+			"plid", String.valueOf(layout.getPlid()));
+
+		String segmentsEntryName = RandomTestUtil.randomString();
+
+		SegmentsEntry segmentsEntry = SegmentsTestUtil.addSegmentsEntry(
+			_group.getGroupId(), RandomTestUtil.randomString(),
+			segmentsEntryName, RandomTestUtil.randomString(), StringPool.BLANK,
+			SegmentsEntryConstants.SOURCE_DEFAULT);
+
+		SegmentsExperience segmentsExperience =
+			SegmentsTestUtil.addSegmentsExperience(
+				segmentsEntry.getSegmentsEntryId(),
+				_classNameLocalService.getClassNameId(Layout.class.getName()),
+				layout.getPlid(),
+				ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+
+		mockLiferayResourceRequest.setParameter(
+			"segmentsExperienceId",
+			String.valueOf(segmentsExperience.getSegmentsExperienceId()));
+
+		_mvcResourceCommand.serveResource(
+			mockLiferayResourceRequest, mockLiferayResourceResponse);
+
+		ByteArrayOutputStream byteArrayOutputStream =
+			(ByteArrayOutputStream)
+				mockLiferayResourceResponse.getPortletOutputStream();
+
+		JSONObject jsonObject = JSONFactoryUtil.createJSONObject(
+			new String(byteArrayOutputStream.toByteArray()));
+
+		JSONObject propsJSONObject = jsonObject.getJSONObject("props");
+
+		String selectedSegmentsExperienceId = propsJSONObject.getString(
+			"selectedSegmentsExperienceId");
+
+		Assert.assertEquals(
+			String.valueOf(segmentsExperience.getSegmentsExperienceId()),
+			selectedSegmentsExperienceId);
```

Comment:
Single use variables should be avoided following Liferay Guidelines: https://liferay.atlassian.net/wiki/spaces/ENGR/pages/1511752519/Java+conventions+in+Liferay#[inlineExtension]Single-use-variables-should-be-avoided

Reply:
done